x-pack/plugin/frozen-indices/src/internalClusterTest/java/org/elasticsearch/index/engine/frozen/FrozenIndexTests.java

```diff
@@ -469,27 +468,25 @@ public void testCanMatch() throws IOException {
                 ).canMatch()
             );
 
-            expectThrows(SearchContextMissingException.class, () -> {
-                ShardSearchContextId withoutCommitId = new ShardSearchContextId(contextId.getSessionId(), contextId.getId(), null);
-                sourceBuilder.query(QueryBuilders.rangeQuery("field").gt("2010-01-06T02:00").lt("2010-01-07T02:00"));
-                assertFalse(
-                    searchService.canMatch(
-                        new ShardSearchRequest(
-                            OriginalIndices.NONE,
-                            searchRequest,
-                            shard.shardId(),
-                            0,
-                            1,
-                            AliasFilter.EMPTY,
-                            1f,
-                            -1,
-                            null,
-                            withoutCommitId,
-                            null
-                        )
-                    ).canMatch()
-                );
-            });
+            ShardSearchContextId withoutCommitId = new ShardSearchContextId(contextId.getSessionId(), contextId.getId(), null);
+            sourceBuilder.query(QueryBuilders.rangeQuery("field").gt("2010-01-06T02:00").lt("2010-01-07T02:00"));
+            assertTrue(
+                searchService.canMatch(
+                    new ShardSearchRequest(
+                        OriginalIndices.NONE,
+                        searchRequest,
+                        shard.shardId(),
+                        0,
+                        1,
+                        AliasFilter.EMPTY,
+                        1f,
+                        -1,
+                        null,
+                        withoutCommitId,
+                        null
+                    )
+                ).canMatch()
+            );
```

Comment:
the previous version of the assertion came from retrying PIT requests on search context missing. I don't think it's a good reason to make can match fail on it though, hence I adjusted the test. Retries will be attempted in the query phase.